### PR TITLE
Add release checklist

### PR DESCRIPTION
**Pull request by briandowns**
Signed-off-by: Brian Downs <brian.downs@gmail.com>

### CODE REVIEW

**Review by manuelbuil — APPROVED** [review]

File: developer-docs/upgrading_kubernetes.md [hunk 1]
```diff
@@ -115,3 +117,20 @@ This step is specific to Rancher and serves to update Rancher's [Kontainer Drive
...
+- [ ] Update RKE2
+- [ ] Tag new RKE2 RC
+- [ ] Tag new RKE2 packaging RC "testing"
+- [ ] Tag the RKE2 release
+- [ ] Tag RKE2 packaging release "testing"
+- [ ] Tag RKE2 packaging release "latest"
+- [ ] Write release notes
+- [ ] Update KDM
+- [ ] Uncheck "pre-release"
+- [ ] Tag RKE2 packaging release "stable"
```
Comment (brandond) [inline developer-docs/upgrading_kubernetes.md:136]:
```suggestion
- [ ] Add release notes to release
- [ ] Tag RKE2 packaging release "testing"
- [ ] Tag RKE2 packaging release "latest"
- [ ] Uncheck "pre-release"
- [ ] Wait 1 week for community to report issues
- [ ] Tag RKE2 packaging release "stable"
- [ ] Update stable release in channels.yaml
- [ ] Update KDM
```

**Review by brandond — CHANGES_REQUESTED** [review]

**Review by brandond — APPROVED** [review]

**Review by dweomer — APPROVED** [review]
LGTM

We should capture this in a issue template like we do for k3s, no?
See https://github.com/k3s-io/k3s/blob/master/.github/ISSUE_TEMPLATE/cut_release.md

**Comment by briandowns** [comment]
> LGTM
> 
> We should capture this in a issue template like we do for k3s, no?
> See https://github.com/k3s-io/k3s/blob/master/.github/ISSUE_TEMPLATE/cut_release.md

Yeah, I could probably get this going next week.

**Review by dereknola — APPROVED** [review]

**Comment by cwayne18** [comment]
Yeah, I think this would be better as an issue template, that's on me for not creating it in the first place.  I think part of the reasoning was that it needed to be different than k3s due to stuff like releasing rpm's, marking them stable, etc.

**Comment by briandowns** [comment]
Would you want this repurposed as a procedural outline in the docs?

**Comment by cwayne18** [comment]
I think that would make sense

**Pull request merged**